### ISSUE TITLE
"Let" declaration interfered with by separate JSX scope

### ISSUE THREAD

**Issue by atombender** [babel/babel#1197]
I can't explain this one. Here's a test case I've been able to whittle down from a bigger class:

```javascript
'use strict';

const Thing = {

  x() {
    return <Dingbat uid={'hello'}/>;
  },

  y() {
    for (let row of this.rows) {
      let uid = 42;
      console.log(uid);
    }
  }
};
```

The generated code (unrelated parts elided):

```javascript
for (var _iterator = this.rows[Symbol.iterator](), _step; !(_iteratorNormalCompletion = (_step = _iterator.next()).done); _iteratorNormalCompletion = true) {
  var row = _step.value;
  var _uid = 42;
  console.log(uid);
}
```

Note how `_uid` is assigned, but it's accessed as `uid`. If I change the JSX line to something else (eg., `<Dingbat bing={'hello'}/>;`) the problem goes away. Seems to me like the JSX scope is leaking into the parallel function scope, and somehow breaking it. I've not been able to reproduce the effect in any other way; replacing the JSX with plain JS code stops the bug from happening, for example.

However, this bug is oddly similar to [this](https://github.com/babel/babel/issues/1169) issue. Same underlying cause, perhaps?

**Comment by sebmck** [comment]
Duplicate of #1169 which I've since fixed with 5.0.11.

**Comment by atombender** [comment]
Confirmed. Thanks for the rapid fix!
